### DISCUSSION TITLE
Request: Support for NavigationBar label alignment on the left of icons

### DISCUSSION THREAD

**Discussion by OmarBelfeki** [flet-dev/flet#4426] [Ideas]
Hello,

In older versions of Flet, I remember being able to position the label directly to the left of the icon in NavigationBarDestination using the default label property. However, in newer versions, this behavior seems to have changed, and the label now appears below the icon by default.

Could you confirm if this was an intentional change? If so, is there any plan to add an option for label alignment (e.g., left, right, top, or bottom of the icon) in future updates?

This feature is essential for my application design, as placing labels to the left of icons provides better usability in some contexts. I’d appreciate any advice, or if there’s a workaround in the current version, I’d be happy to implement it.

Thank you for your help!

**Comment by FeodorFitsner** [comment]
NavigationBar had never allowed placing labels on a side. Maybe you mean NavigationRail?

<img width="506" alt="image" src="https://github.com/user-attachments/assets/1a8a4ef8-2f91-40e8-a4d8-482b35ebff7b">

**Reply by OmarBelfeki** [reply]
i mean like this
![naviga](https://github.com/user-attachments/assets/43641ff2-6dc0-4e32-a1a5-7a55ff5f505e)
